### PR TITLE
[#242] No scan symlinks as md files

### LINKED ISSUE #242: [BUG] Some symlinks are scanned when not necessary
# Description

Looks like we follow symlinks and scan the files that should not be scanned.

# To Reproduce

Have three files:

* `/dir/a`
* `/dir/b.md` that contains `[link](a)` link.
* `/s.md` -> `/dir/b.md` symlink.

This results in an error: `s.md` refers to a non-existing `a` file.

# Expected behavior

At least in GitLab we should not show such error - GitLab does not try to follow and render symlinks, so we should not treat `s.md` as a markdown document rather as an empty file I suppose. (Or maybe we could verify that the symlink is valid :thinking:)

I don't know what happens in GitHub, but even if it tries to render symlinks to documentation and this results into a bunch of broken links, I think we should be smart here and report only one error saying that the rendered symlink will have all broken references.

# Environment

- GitLab
- 0.2.2

### PR DESCRIPTION
**Pull request by aeqz**
Problem: when the repository contains a symlink to a markdown file, it is processed by xrefcheck as if it was the same markdown file but in the symlink's location. This leads to broken references and can be avoided because neither GitHub nor GitLab try to render symlinks as the file they point to.

Solution: consider symlinks as no scannable files. In the future, we will consider to include a new dedicated scanner for symlinks if it works.

## Related issue(s)

Fixes #242
Relates #244

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](https://github.com/serokell/xrefcheck/tree/master/README.md)
    - Haddock

- Public contracts
  - [x] Any modifications of public contracts comply with the [Evolution
  of Public Contracts](https://www.notion.so/serokell/Evolution-of-Public-Contracts-2a3bf7971abe4806a24f63c84e7076c5) policy.
  - [x] I added an entry to the [changelog](https://github.com/serokell/xrefcheck/tree/master/CHANGES.md) if my changes are visible to the users
        and
  - [x] provided a migration guide for breaking changes if possible

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [x] My code complies with the [style guide](https://github.com/serokell/style/blob/master/haskell.md).

### CODE REVIEW

**Comment by aeqz** [comment]
After several attempts on where to ignore symlinks. I have implemented the simplest one and we will complicate it in a following PR for #244 if necessary.

File: CHANGES.md [hunk 1]
```diff
@@ -44,10 +44,11 @@ Unreleased
 * [#254](https://github.com/serokell/xrefcheck/pull/254)
   + Now the `dump-config` command does not overwrite a file unless explicitly told with a
     `--force` flag. Also, a `--stdout` flag allows to print the config to stdout instead.
-    the configured Markdown flavour.
```
Comment (Martoon-00) [inline CHANGES.md:47]: That's a good change :+1: 

And I would like it to be placed to a different commit :smile_cat:

**Review by Martoon-00 — COMMENTED** [review]
Thanks, looks good to me.

I extra checked the commit description, and see everything I wanted to see there :+1: 

Leaving just two comments.

**Review by YuriRomanowski — APPROVED** [review]

File: tests/golden/check-symlinks/check-symlinks.bats [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bats
+
+# SPDX-FileCopyrightText: 2022 Serokell <https://serokell.io>
```
Comment (Martoon-00) [inline tests/golden/check-symlinks/check-symlinks.bats:3]: :warning: The year in this license header is outdated, time to update!





<p align="right" data-meta="generated_by_instant-checks">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: tests/golden/check-symlinks/dir/b.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<!--
+ - SPDX-FileCopyrightText: 2022 Serokell <https://serokell.io>
```
Comment (Martoon-00) [inline tests/golden/check-symlinks/dir/b.md:2]: :warning: The year in this license header is outdated, time to update!





<p align="right" data-meta="generated_by_instant-checks">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by Martoon-00** [comment]
:warning: The year in this license header is outdated, time to update!



<p align="right" data-meta="generated_by_instant-checks">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: CHANGES.md (hunk 1 above)
Comment (Martoon-00) [reply CHANGES.md:47]: And the addition below should probably belong to the main commit )

Also, I would add a description to that commit you extracted after all. At a first glance, it's not clear why exactly you delete this line, maybe it was meant to contain some important information and should be extended, not removed. If in the commit description you mention that this was an accidental copy-paste, that would make the reasoning much clearer for an arbitrary git history reader.

Comment (Martoon-00) [reply CHANGES.md:47]: Very good, thank you :+1:
